1-server-side/msal-webapi-ciam-sample/msal-web-api/src/main/java/com/microsoft/azure/msalapiciamsample/AADClaimsVerifier.java

```diff
@@ -0,0 +1,67 @@
+package com.microsoft.azure.msalapiciamsample;
+
+import org.springframework.security.oauth2.common.exceptions.InvalidTokenException;
+import org.springframework.security.oauth2.provider.token.store.JwtClaimsSetVerifier;
+import org.springframework.util.Assert;
+import org.springframework.util.CollectionUtils;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Map;
+
+public class AADClaimsVerifier implements JwtClaimsSetVerifier {
+    private static final String ISS_CLAIM = "iss";
+    private static final String AUD_CLAIM = "aud";
+    private static final String V2_ISSUER_FORMAT = "https://%s.ciamlogin.com/%s/v2.0";
+
+
+    private HashSet<String> acceptedIssuers = new HashSet<>();
+    private final String applicationId;
+
+    public AADClaimsVerifier(final String[] acceptedTenants, final String applicationId) {
+
+        Assert.notEmpty(acceptedTenants, "acceptedTenants cannot be empty");
```

Comment:
What does Assert do in product code?

Reply:
Yes, [assertions](https://docs.oracle.com/javase/8/docs/technotes/guides/language/assert.html) will thrown an exception if the expression inside isn't true. They're mostly used as an easy to write sanity check to say "if this ever gets thrown there's an obvious bug", in this case it means a major issue with the config file.